### PR TITLE
refactor(j-s): Case Completion Tasks

### PR DESCRIPTION
**Pull request by gudjong**
# Case Completion Tasks

[Nota queue þegar verið er að ljúka máli](https://app.asana.com/0/1199153462262248/1202505428021062/f)

## What

- Moves all tasks previously performed after ruling signature to a downstream queuing task, except storing the signed ruling in S3.

## Why

Makes the client more responsive and the backend more recilient.

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: apps/judicial-system/backend/src/app/modules/notification/notification.service.ts [hunk 1]
```diff
@@ -669,9 +678,15 @@ export class NotificationService {
 
   private async sendCourtDateNotifications(
     theCase: Case,
-    user: User,
+    user?: User,
     eventOnly?: boolean,
   ): Promise<SendNotificationResponse> {
+    if (!user) {
```
Comment (Arnarkari93) [inline apps/judicial-system/backend/src/app/modules/notification/notification.service.ts:684]: Is this needed? Why did we change the user from being a required param?

File: apps/judicial-system/backend/src/app/modules/notification/notification.config.ts [hunk 2]
```diff
@@ -0,0 +1,39 @@
+import { defineConfig } from '@island.is/nest/config'
+
+export const notificationModuleConfig = defineConfig({
+  name: 'NotificationModule',
+  load: (env) => ({
+    production: env.optional('NODE_ENV') === 'production',
+    email: {
+      fromEmail: env.required('EMAIL_FROM', 'ben10@omnitrix.is'),
```
Comment (Arnarkari93) [inline apps/judicial-system/backend/src/app/modules/notification/notification.config.ts:8]: I am just curious. I see that this email is currently being used in the env config. What email is this?

File: apps/judicial-system/backend/src/app/modules/notification/notification.service.ts [hunk 3]
```diff
@@ -816,8 +882,14 @@ export class NotificationService {
 
   private async sendModifiedNotifications(
     theCase: Case,
-    user: User,
+    user?: User,
   ): Promise<SendNotificationResponse> {
+    if (!user) {
```
Comment (Arnarkari93) [inline apps/judicial-system/backend/src/app/modules/notification/notification.service.ts:887]: Same here, why change the user from being required?

**Review by Arnarkari93 — COMMENTED** [review]
I started reviewing but I think I would prefer if we review this PR together to get a better grasp of what is happening.

File: apps/judicial-system/backend/src/app/modules/notification/notification.service.ts (hunk 1 above)
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/notification/notification.service.ts:684]: The notification service can now be called by initiated by the message handler where there is no user involved.

File: apps/judicial-system/backend/src/app/modules/notification/notification.config.ts (hunk 2 above)
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/notification/notification.config.ts:8]: This is something devops added a long time ago, some "valid" dummy email that the build system does not complain about. Real email addresses are not allowed in the repository.

File: apps/judicial-system/backend/src/app/modules/notification/notification.service.ts (hunk 3 above)
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/notification/notification.service.ts:887]: Same reason.

File: apps/judicial-system/backend/src/app/modules/case/case.service.ts [hunk 4]
```diff
@@ -358,192 +345,118 @@ export class CaseService {
...
-                courtRecordUploadedToCourt = res
-              })
-            })
+  private async deliverSignedRulingToCourt(theCase: Case): Promise<boolean> {
+    return this.awsS3Service
+      .getObject(`generated/${theCase.id}/ruling.pdf`)
+      .then((pdf) =>
+        this.uploadSignedRulingPdfToCourt(theCase, pdf).catch(() => false),
+      )
+      .catch(() => true) // The signed ruling should have been delivered by email
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/case/case.service.ts]: Done

File: apps/judicial-system/backend/src/app/modules/case/internalCase.controller.ts [hunk 5]
```diff
@@ -43,4 +54,19 @@ export class InternalCaseController {
...
+  @Post('case/:caseId/deliver')
+  @ApiOkResponse({
+    type: DeliverResponse,
+    description: 'Delivers a completed case to court',
+  })
+  deliver(
+    @Param('caseId') _caseId: string,
+    @CurrentCase() theCase: Case,
+  ): Promise<DeliverResponse> {
+    this.logger.debug('Delivering a case')
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/backend/src/app/modules/case/internalCase.controller.ts:68]: done

File: apps/judicial-system/web/src/components/SigningModal/SigningModal.tsx [hunk 6]
```diff
@@ -203,32 +175,42 @@ const SigningModal: React.FC<SigningModalProps> = ({
...
           ? undefined
           : signingProgress === 'success'
           ? 'Loka glugga'
-          : 'Loka og reyna aftur'
-      }
-      primaryButtonText={
-        data?.rulingSignatureConfirmation?.documentSigned
-          ? 'Senda ábendingu'
-          : ''
+          : 'Reyna aftur'
```
[the remark this answers is not in the document]
Comment (gudjong) [reply apps/judicial-system/web/src/components/SigningModal/SigningModal.tsx:190]: Will create a separate Asana task for this.

**Review by Arnarkari93 — APPROVED** [review]

**Pull request merged**